apisix/plugin.lua

```diff
@@ -655,7 +655,8 @@ function _M.merge_consumer_route(route_conf, consumer_conf, api_ctx)
     core.log.info("route conf: ", core.json.delay_encode(route_conf))
     core.log.info("consumer conf: ", core.json.delay_encode(consumer_conf))
 
-    local flag = tostring(route_conf) .. tostring(consumer_conf)
+    local flag = route_conf.value.id .. "#" .. route_conf.modifiedIndex
+                 .. consumer_conf.id .. "#" .. consumer_conf.modifiedIndex
```

Comment:
Better to add a `#` between route_conf.modifiedIndex and consumer_conf.id

Reply:
done